tpm2tools/seal_test.go

```diff
@@ -15,39 +16,52 @@ func TestSeal(t *testing.T) {
 	rwc := internal.GetTPM(t)
 	defer CheckedClose(t, rwc)
 
-	key, err := StorageRootKeyRSA(rwc)
-	if err != nil {
-		t.Fatalf("can't create srk from template: %v", err)
-	}
-	defer key.Close()
-
-	secret := []byte("test")
-	pcrList := []int{7, 23}
-	pcrToExtend := tpmutil.Handle(23)
-
-	sealed, err := key.Seal(pcrList, secret)
-	if err != nil {
-		t.Fatalf("failed to seal: %v", err)
-	}
-
-	unseal, err := key.Unseal(sealed)
-	if err != nil {
-		t.Fatalf("failed to unseal: %v", err)
-	}
-	if !bytes.Equal(secret, unseal) {
-		t.Fatalf("unsealed (%v) not equal to secret (%v)", unseal, secret)
-	}
-
-	extension := bytes.Repeat([]byte{0xAA}, sha256.Size)
-	err = tpm2.PCRExtend(rwc, pcrToExtend, tpm2.AlgSHA256, extension, "")
-	if err != nil {
-		t.Fatalf("failed to extend pcr: %v", err)
-	}
-
-	// unseal should not succeed.
-	_, err = key.Unseal(sealed)
-	if err == nil {
-		t.Fatalf("unseal should have caused an error: %v", err)
+	tests := []struct {
+		name   string
+		getSRK func(io.ReadWriter) (*Key, error)
+	}{
+			{"RSA", StorageRootKeyRSA},
+			{"ECC", StorageRootKeyECC},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			srk, err := test.getSRK(rwc)
+
+			if err != nil {
+				t.Fatalf("can't create %s srk from template: %v", test.name, err)
+			}
+			defer srk.Close()
+
+			secret := []byte("test")
+			pcrList := []int{7, 23}
+			pcrToExtend := tpmutil.Handle(23)
+		
+			sealed, err := srk.Seal(pcrList, secret)
+			if err != nil {
+				t.Fatalf("failed to seal: %v", err)
+			}
+		
+			unseal, err := srk.Unseal(sealed)
+			if err != nil {
+				t.Fatalf("failed to unseal: %v", err)
+			}
+			if !bytes.Equal(secret, unseal) {
+				t.Fatalf("unsealed (%v) not equal to secret (%v)", unseal, secret)
+			}
+		
+			extension := bytes.Repeat([]byte{0xAA}, sha256.Size)
+			err = tpm2.PCRExtend(rwc, pcrToExtend, tpm2.AlgSHA256, extension, "")
+			if err != nil {
+				t.Fatalf("failed to extend pcr: %v", err)
+			}
+		
+			// unseal should not succeed.
+			_, err = srk.Unseal(sealed)
```

Comment:
Move this into the if statement (I realize that this is how it was before, but we want to be consistent).